cmd/agentbasedinstaller/register.go

```diff
@@ -49,7 +50,7 @@ func RegisterCluster(ctx context.Context, log *log.Logger, bmInventory *client.A
 	}
 	clusterResult, registerClusterErr := bmInventory.Installer.V2RegisterCluster(ctx, clientClusterParams)
 	if registerClusterErr != nil {
-		return nil, registerClusterErr
+		return nil, errorutil.GetAssistedError(registerClusterErr)
```

Comment:
I did some testing and I am not seeing the improved error reporting. I think errorutil.GetAssistedError is not returning the detailed error message because the errors type check is flowing into the default case here: https://github.com/openshift/assisted-service/blob/master/pkg/error/error_utils.go#L44

Reply:
It looks like you ran into a local error and thus didn't get the response from the server:
https://github.com/openshift/assisted-service/blob/master/client/installer/v2_register_cluster_responses.go#L63

Reply:
I was able to trigger a non-local error and now seeing the extra information in the error message. The trigger was to deploy a SNO cluster with API_VIP and INGRESS_VIP defined in the AgentClusterInstall.

`Jul 07 05:30:13 master-0.ostest.test.metalkube.org create-cluster-and-infraenv[2668]: time="2022-07-07T05:30:13Z" level=fatal msg="Failed to register cluster with assisted-service: AssistedServiceError Code: 400 Href:  ID: 400 Kind: Error Reason: Ingress VIP cannot be set with User Managed Networking"`